test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightBatchReaderTests.cs

```diff
@@ -363,6 +363,114 @@ public async Task ReadBatchRequestWithChangesetAndDependsOnIdsAsync()
+                        {
+                            if (jsonLightBatchReader.State == ODataBatchReaderState.Operation)
+                            {
+                                // The json properties are just iterated through and have no purpose for this test case
+                                await jsonLightBatchReader.CreateOperationRequestMessageAsync();
+                            }
+                        }
+                        Assert.False(true, "The test failed, because the duplicate header has thrown no ODataException");
+                    }
+                    catch (ODataException ex)
+                    {
+                        // Verify that the correct duplicate property has raised the ODataException
+                        Assert.Equal(Strings.ODataJsonLightBatchPayloadItemPropertiesCache_DuplicatePropertyForRequestInBatch("ATOMICITYGROUP"), ex.Message);
+                    }
+                },
+                isResponse: false);
+        }
+
+        [Fact]
+        public async Task ReadBatchRequestWithDuplicateHeadersAsync()
+        {
+            var payload = "{\"requests\": [{" +
+                "\"id\": \"1\"," +
+                "\"method\": \"POST\"," +
+                "\"url\": \"http://tempuri.org/Customers\"," +
+                "\"headers\": {\"odata-version\":\"4.0\",\"content-type\":\"application/json;odata.metadata=minimal;odata.streaming=true;IEEE754Compatible=false;charset=utf-8\",\"duplicate-header\":\"value1\",\"duplicate-header\":\"value2\"}, " +
+                "\"body\": {\"@odata.type\":\"#NS.Customer\",\"Id\":1,\"Name\":\"Customer 1\",\"Type\":\"Retail\"}}]}";
+
+            await SetupJsonLightBatchReaderAndRunTestAsync(
+                payload,
+                async (jsonLightBatchReader) =>
+                {
+                    try
+                    {
+                        while (await jsonLightBatchReader.ReadAsync())
+                        {
+                            if (jsonLightBatchReader.State == ODataBatchReaderState.Operation)
+                            {
+                                // The json properties are just iterated through and have no purpose for this test case
+                                await jsonLightBatchReader.CreateOperationRequestMessageAsync();
+                            }
+                        }
+                        Assert.False(true, "The test failed, because the duplicate header has thrown no ODataException");
+                    }
+                    catch (ODataException ex)
+                    {
+                        // Verify that the correct duplicate header has raised the ODataException
+                        Assert.Equal(Strings.ODataJsonLightBatchPayloadItemPropertiesCache_DuplicateHeaderForRequestInBatch("duplicate-header"), ex.Message);
+                    }
+                },
+                isResponse: false);
+        }
+

```

Comment:
Somewhere within this test project we have tests for the sync API. We should add tests for the sync scenarios as well.

Reply:
Unfortunately, I've not found any matching sync `ODataBatchReader` UnitTests, that matches to these use-cases.
There are test cases, where the sync methods are called, e.g. `ODataMultipartMixedBatchReaderTests`, but these focus on different topics, so it doesn't make sense to add null and duplicate header verifications to them.
If for these test scanarios dedicated sync method UnitTests are required / desired, it would be a topic for an upcomming feature ticket.

Reply:
We strive to ensure that the sync and async APIs do not diverge so this ask is important... We have to ensure that the same exceptions that are thrown in the async workflow will be thrown in the sync workflow.

Reply:
I've added 3x synchronous unit tests to verify that `null` headers and duplicate properties & headers.
If you wish further improvements to unit test coverage, please feel free to raise an independent feature ticket, since I won't cover other non existing tests, that don't belong to this bugfix.